plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1341,12 +1344,13 @@ public OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandle
 
         Table icebergTable = catalog.loadTable(session, handle.getSchemaTableName());
 
-        icebergTable.newDelete()
-                .deleteFromRowFilter(toIcebergExpression(handle.getEnforcedPredicate()))
+        DeleteFiles deleteFiles = icebergTable.newDelete()
+                .deleteFromRowFilter(toIcebergExpression(handle.getEnforcedPredicate()));
+        long deletedRecords = Long.parseLong(deleteFiles.apply().summary().getOrDefault(DELETED_RECORDS_PROP, "0"));
```

Comment:
When can the information be missing?

if it can actually be missing, we should return empty instead of 0,
but maybe it's guaranteed to exist and then we just should get and check?

Reply:
Are you sure? I actually think that it is better to return 0,  it is more consistent for me when all deletes return information in the same way. Even tests show this as in current state this passes:
`assertUpdate("DELETE FROM " + tableName + " WHERE key = 'three'", 0);`
and it won't pass with the proposed change.
even this will fail:
`assertQueryReturnsEmptyResult("DELETE FROM " + tableName + " WHERE key = 'three'");` 
with:
`java.lang.AssertionError: [Rows for query [DELETE FROM test_delete_returns_number_of_rows_1npq8u144q WHERE key = 'three']] 
Expecting empty but was:<[[null]]>`

@findepi WDYT?

Reply:
> When can the information be missing?

When nothing is deleted this property is not set.

Reply:
> information in the same way. Even tests show this as in current state this passes:
assertUpdate("DELETE FROM " + tableName + " WHERE key = 'three'", 0);
and it won't pass with the proposed change.

use `assertUpdate("DELETE...");` (remove `, 0`) there

> When nothing is deleted this property is not set.

Sounds like something that we should be possible to fix.

Until it's fixed, i don't want to rely on a quirk that lack of DELETED_RECORDS_PROP means "0 rows affected".
That sounds like relying on an impl detail.